agent/engine/docker_task_engine.go

```diff
@@ -1924,6 +1927,15 @@ func (engine *DockerTaskEngine) createContainer(task *apitask.Task, container *a
 		}
 	}
 
+	// If the container has mode=non-blocking set, and the max-buffer-size field is
+	// unset, then default to 10m.
```

Comment:
nit: maybe we should not say `10m` here so that the possibility of someone changing the constant in the future and forgetting to update this comment is removed. We can add a comment for the constant definition?